CDCgov/prime-reportstream
# Implement feature flags v2 backend support

This PR reverts the changes made in https://github.com/CDCgov/prime-reportstream/pull/7555 and implements the "version 2" of feature flags as discussed in prior meetings and in the description of epic https://github.com/cdcgov/prime-reportstream/issues/7734. This new implementation relies on a new "virtual table" (aka lookup table) named rs_feature_flag to store feature flags and adds a featureFlags property to the Organization object to allow the assignment of feature flags to organizations. 

## Test Steps:
- Run FlyWay migration and confirm it deletes the two permission tables: `gradelw clean package`
- Manually create `rs_feature_flag` lookup table and verify DB looks correct:
  - `./prime lookuptables create -e local -n rs_feature_flag -i metadata/tables/local/rs_feature_flag.csv `
- Add a feature to an org
  - `./prime organization set -e local -i ~/Documents/assets/feature_org.yml `
  - Verify feature was added to org by reading back the org
    - `./prime organization get -e local -n feature_org `:
      
      ```
			name: "feature_org"
			description: "an org for testing the feature flags feature"
			jurisdiction: "FEDERAL"
			filters:
			- topic: "covid-19"
			  jurisdictionalFilter:
			  - "matches(ordering_facility_state, IG)"
			  qualityFilter: null
			  routingFilter: null
			  processingModeFilter: null
			- topic: "monkeypox"
			  jurisdictionalFilter:
			  - "matches(ordering_facility_state, IG)"
			  qualityFilter: null
			  routingFilter: null
			  processingModeFilter: null
			- topic: "full-elr"
			  jurisdictionalFilter:
			  - "%performerState = 'IG' or %patientState = 'IG'"
			  qualityFilter:
			  - "%hl7MessageType = 'R01'"
			  routingFilter: null
			  processingModeFilter: null
			featureFlags:
			- "message-tracker"
			version: 0
			createdBy: "local@test.com"
			createdAt: "2023-01-10T16:31:57.35Z"
      ```



## Changes
- Reverted changes in https://github.com/CDCgov/prime-reportstream/pull/7555
  - Implemented FlyWay migration to remove permission tables
- Created .csv to create initial feature flags
- Updated organization object to have featureFlags property

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #7737

## To Be Done
- Experience team needs to add "front-end" stories to https://github.com/cdcgov/prime-reportstream/issues/7734 to capture the work they will need to do to complete the feature flags feature (I added some hints of what needs to be done in the epic description).
- Once this is merged, I will need to manually run the FlyWay migration on Production and Staging DBs and initialize the featureFlags lookup table
